Add files via upload

networkers-pl opened the pull request:
This is a modified copy of the cisco_spark.py file. The name change is associated with the change of the product name from Cisco Spark to Cisco Webex Teams. In addition, the current version (cisco_spark) does not work due to a name collision with MESSAGE. I had to modify the name from "message" to "webexmsg" in many places in cisco_webex. It works fine in this version.

The original author has been preserved and copied from cisco_spark to cisco_webex.
@drew-russell

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

The current version of cisco_spark module does not work due to a name collision with MESSAGE. I had to modify the name from "message" to "webexmsg" in many places in cisco_webex.py file. It works fine in this version.

The name change is associated with the change of the product name from Cisco Spark to Cisco Webex Teams. 

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request
- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
cisco_webex

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+short_description: Send a webexmsg to a Cisco Webex Teams Room or Individual.
+description:
+    - Send a webexmsg to a Cisco Webex Teams Room or Individual with options to control the formatting.
+version_added: "2.3"
```
```suggestion
version_added: "2.10"
```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+    description:
+       - The request parameter you would like to send the webexmsg to.
+       - Messages can be sent to either a room or individual (by ID or E-Mail).
+    required: True
```
```suggestion
    required: yes
```
etc. for the other parameters
it's from the doc

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+    description:
+      - Your personal access token required to validate the Webex Teams API.
+    required: true
+    aliases: ['token']
```
```suggestion
```
better to avoid extra aliases

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+  webexmsg_type:
+    description:
+       - Specifies how you would like the webexmsg formatted.
+    required: False
```
```suggestion
```
it's the default

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+    webexmsg_type: text
+    personal_token: "{{ token }}"
+    webexmsg: "Cisco Webex Teams Ansible Module - Text Message to Individual by E-Mail"
+
```
```suggestion
```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+
+
+def spark_webexmsg(module):
+    """ When check mode is specified, establish a read only connection, that does not return any user specific
```
```suggestion
    """When check mode is specified, establish a read only connection, that does not return any user specific
```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+            webexmsg_type=dict(required=False, default=['text'], aliases=[
+                              'type'], choices=['text', 'markdown']),
+            personal_token=dict(required=True, no_log=True, aliases=['token']),
+            webexmsg=dict(required=True)
+
```
```suggestion
            webexmsg=dict(required=True),
```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+            recipient_id=dict(required=True, no_log=True),
+            webexmsg_type=dict(required=False, default=['text'], aliases=[
+                              'type'], choices=['text', 'markdown']),
+            personal_token=dict(required=True, no_log=True, aliases=['token']),
```
```suggestion
            personal_token=dict(required=True, no_log=True),
```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+            recipient_type=dict(required=True, choices=[
+                'roomId', 'toPersonEmail', 'toPersonId']),
+            recipient_id=dict(required=True, no_log=True),
+            webexmsg_type=dict(required=False, default=['text'], aliases=[
+                              'type'], choices=['text', 'markdown']),
```
```suggestion
            webexmsg_type=dict(required=False, default=['text'], choices=['text', 'markdown']),
```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+    '''Ansible main. '''
+    module = AnsibleModule(
+        argument_spec=dict(
+            recipient_type=dict(required=True, choices=[
+                'roomId', 'toPersonEmail', 'toPersonId']),
```
```suggestion
            recipient_type=dict(required=True, choices=['roomId', 'toPersonEmail', 'toPersonId']),
```

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+
```
```suggestion
```
No longer needed (it must be removed)

Andersson007 commented on plugins/modules/notification/cisco_webex.py:
```diff
@@ -0,0 +1,192 @@
+DOCUMENTATION = '''
+---
+module: cisco_webex
+short_description: Send a webexmsg to a Cisco Webex Teams Room or Individual.
```
```suggestion
short_description: Send a webexmsg to a Cisco Webex Teams Room or Individual
```
"Unlike description:, short_description should not have a trailing period/full stop." from the [doc](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block)

Andersson007 commented:
@networkers-pl thanks for the PR!

also the sanity errors need to be fixed like
```plugins/modules/notification/cisco_webex.py:177:31: E126: continuation line over-indented for hanging indent```

felixfontein commented:
You also need to create a symlink from `plugins/modules/` to your module.

networkers-pl commented:
I understand that I should do it. I will look at it soon. 
I am asking you for your understanding, I have never used GitHub before and I am very lost in it.


> On 22 May 2020, at 09:21, Andrew Klychkov <notifications@github.com> wrote:
> 
> 
> @networkers-pl <https://github.com/networkers-pl> thanks for the PR!
> 
> also the sanity errors need to be fixed like
> plugins/modules/notification/cisco_webex.py:177:31: E126: continuation line over-indented for hanging indent
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/ansible-collections/community.general/pull/393#issuecomment-632537211>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APVNUU4T2WYLBKG5UFVMXNLRSYRYPANCNFSM4NG5S3PQ>.
>

networkers-pl commented:
Can I really improve it myself and it doesn't have to be done by the original author Drew Rusell (@drew-russell)?

Sorry, but I do not understand this suggestion.
I corrected the rest of them.

> On 22 May 2020, at 09:19, Andrew Klychkov <notifications@github.com> wrote:
> 
> In plugins/modules/notification/cisco_webex.py:
> 
> > +- name: Cisco Webex Teams - Text Message by an Individuals ID
> +  cisco_webex:
> +    recipient_type: toPersonId
> +    recipient_id: "{{ person_id}}"
> +    webexmsg_type: text
> +    personal_token: "{{ token }}"
> +    webexmsg: "Cisco Webex Teams Ansible Module - Text Message to Individual by ID"
> +
> +- name: Cisco Webex Teams - Text Message by an Individuals E-Mail Address
> +  cisco_webex:
> +    recipient_type: toPersonEmail
> +    recipient_id: "{{ person_email }}"
> +    webexmsg_type: text
> +    personal_token: "{{ token }}"
> +    webexmsg: "Cisco Webex Teams Ansible Module - Text Message to Individual by E-Mail"
> +
> 
> ⬇️ Suggested change
> -

networkers-pl commented:
I corrected everything in accordance with the e-mails and I already have it in my repository/fork. 
I am looking the way to send it, so it appears in your repository, so it will be soon. 
I'm sorry that it took so long. 

> On 22 May 2020, at 09:27, Felix Fontein <notifications@github.com> wrote:
> 
> 
> You also need to create a symlink from plugins/modules/ to your module.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/ansible-collections/community.general/pull/393#issuecomment-632539970>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APVNUUYMU5DD56QDHBT4JXTRSYSPNANCNFSM4NG5S3PQ>.
>

networkers-pl commented:
Corrections after sanity errors and mail suggestions and symlink.

networkers-pl commented:
I think it should already be visible on your side. if not, please let me know.

> On 22 May 2020, at 12:41, Marcin Sleczek <marcin.sleczek@networkers.pl> wrote:
> 
> I corrected everything in accordance with the e-mails and I already have it in my repository/fork. 
> I am looking the way to send it, so it appears in your repository, so it will be soon. 
> I'm sorry that it took so long. 
> 
>> On 22 May 2020, at 09:27, Felix Fontein <notifications@github.com <mailto:notifications@github.com>> wrote:
>> 
>> 
>> You also need to create a symlink from plugins/modules/ to your module.
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub <https://github.com/ansible-collections/community.general/pull/393#issuecomment-632539970>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APVNUUYMU5DD56QDHBT4JXTRSYSPNANCNFSM4NG5S3PQ>.
>> 
>

Andersson007 commented:
Why have you closed the PR? you just need to add `type: str`, etc. for every parameter in the documentation block (the same types as listed in `argument_spec`. Looks like it's an only problem with the sanity tests.
When you solve a note please push "Resolve conversation" button.

networkers-pl commented:
Sorry for that. I didn't have time to read carefully about GitHub and just started correcting everything right away.
I don't know much about what happens when I click something.. and I don't remember yet what I clicked earlier.

I did as you requested and I understood. 
I am now starting to test this module in practice after these changes.

If something else is wrong, please let me know.


> On 22 May 2020, at 13:02, Andrew Klychkov <notifications@github.com> wrote:
> 
> 
> Why have you closed the PR? you just need to add type: str, etc. for every parameter in the documentation block (the same types as listed in argument_spec. Looks like it's an only problem with the sanity tests.
> When you solve a note please push "Resolve conversation" button.
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/ansible-collections/community.general/pull/393#issuecomment-632633302>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APVNUU45L646MZDQYD7FUW3RSZLULANCNFSM4NG5S3PQ>.
>

Andersson007 commented:
It's closed now. You can reopen it.
In general, it's not a good idea to create backports in master, so
1. copy these changes with your local repository to somewhere
2. remove your forked repo on GitHub
3. fork it again
4. clone to your local system
5. create a new branch from master there, e.g. `git checkout -b add_files_via_upload`
6. copy your changes to there, commit, `git push origin add_files_via_upload`
7. create a new PR instead

networkers-pl commented:
Thank you very much for the instruction, it really helped.

I tested the module and it works. 
On May 12, 2020, the API URL also changed. 
The old one still works, but now instead of api.ciscospark.com, you should use webexapis.com.
I also added it to the changes.

More details can be found here:
https://developer.webex.com/blog/introducing-the-new-webexapis-com

Also the old links from documentation section didn't work, as Spark is no longer there. 
I changed them to other Cisco Webex.

I've made all the corrections, I hope that's good.

I think it should be visible by now on your side.


> On 22 May 2020, at 13:34, Andrew Klychkov <notifications@github.com> wrote:
> 
> 
> It's closed now. You can reopen it.
> In general, it's not a good idea to create backports in master, so
> 
> 	• copy these changes with your local repository to somewhere
> 	• remove your forked repo on GitHub
> 	• fork it again
> 	• clone to your local system
> 	• create a new branch from master there, e.g. git checkout -b add_files_via_upload
> 	• copy your changes to there, commit, git push origin add_files_via_upload
> 	• create a new PR instead
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

Andersson007 commented:
@networkers-pl i don't see any changes.
The pr is closed. You need to reopen it or create a new one instead

networkers-pl commented:
Everything should be in # 401.

https://github.com/ansible-collections/community.general/pull/401 <https://github.com/ansible-collections/community.general/pull/401>


> On 23 May 2020, at 09:06, Andrew Klychkov <notifications@github.com> wrote:
> 
> 
> @networkers-pl <https://github.com/networkers-pl> i don't see any changes.
> The pr is closed. You need to reopen it or create a new one instead
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/ansible-collections/community.general/pull/393#issuecomment-632998659>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APVNUU3X7LQHZTANOWV7Y5LRS5YYLANCNFSM4NG5S3PQ>.
>

networkers-pl commented:
Hello. Are you waiting with something for me? Should I close PR or do something else in #401? 


> On 23 May 2020, at 15:15, Marcin Sleczek <marcin.sleczek@networkers.pl> wrote:
> 
> Everything should be in # 401.
> 
> https://github.com/ansible-collections/community.general/pull/401
> 
> 
>> On 23 May 2020, at 09:06, Andrew Klychkov <notifications@github.com> wrote:
>> 
>> 
>> @networkers-pl i don't see any changes.
>> The pr is closed. You need to reopen it or create a new one instead
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub, or unsubscribe.
>> 
>
